### PR TITLE
Fix e2e test for render cmd

### PR DESCRIPTION
**Pull request by Tal-or**
- render-command-tests should depend on dist
- set more clear error message in case test failed
- manual_tuned.yaml's data field should be present as single line

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
this fixes the current e2e tests, however is very hard to understand this is the correct fix without trying it out.
The e2e tests give us an output like this when failing:
```
1011 20:31:04 fromani@musashi2 ~/go/src/github.com/openshift-kni/performance-addon-operators $ make render-command-tests
Running Render Command Tests
hack/run-render-command-functests.sh
/home/fromani/go/bin/ginkgo
Running Functional Tests: functests-render-command
Running Suite: Performance Operator render tests
================================================
Random Seed: 1621794667
Will run 2 of 2 specs

using binary at "/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/build/_output/bin"
render command unittest With a single performance-profile 
  Gets cli args and produces the expected components to output directory
  /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:32
assets` output dir at: "/tmp/assets890611869"
running: [/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/build/_output/bin/performance-addon-operators render --performance-profile-input-files /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/cluster-setup/manual-cluster/performance/performance_profile.yaml --asset-input-dir /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/build/assets --asset-output-dir /tmp/assets890611869]
reference data at: "/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/testdata/render-expected-output"

• Failure [0.021 seconds]
render command unittest
/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:22
  With a single performance-profile
  /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:31
    Gets cli args and produces the expected components to output directory [It]
    /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:32

    Expected
        <int>: 1
    to be zero-valued

    /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:100
------------------------------
render command unittest With a single performance-profile 
  Gets cli args and produces the expected components to output directory
  /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:32
assets` output dir at: "/tmp/assets787232600"
running: [/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/build/_output/bin/performance-addon-operators render --performance-profile-input-files /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/cluster-setup/manual-cluster/performance/performance_profile.yaml --asset-input-dir /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/build/assets --asset-output-dir /tmp/assets787232600]
reference data at: "/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/testdata/render-expected-output"

• Failure [0.019 seconds]
render command unittest
/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:22
  With a single performance-profile
  /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:31
    Gets cli args and produces the expected components to output directory [It]
    /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:32

    Expected
        <int>: 1
    to be zero-valued

    /home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:100
------------------------------
S
JUnit report was created: /tmp/artifacts/unit_report_render_manifests.xml


Summarizing 2 Failures:

[Fail] render command unittest With a single performance-profile [It] Gets cli args and produces the expected components to output directory 
/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:100

[Fail] render command unittest With a single performance-profile [It] Gets cli args and produces the expected components to output directory 
/home/fromani/go/src/github.com/openshift-kni/performance-addon-operators/functests-render-command/1_render_command/render_test.go:100

Ran 1 of 2 Specs in 0.041 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Flaked | 0 Pending | 1 Skipped
--- FAIL: TestRenderCmd (0.04s)
FAIL

Ginkgo ran 1 suite in 600.075594ms
Test Suite Failed
make: *** [Makefile:268: render-command-tests] Error 1
```

which makes hard to understand the failure reason.
We should improve also the test message (perhaps in another PR).

File: Makefile [hunk 1]
```diff
@@ -263,7 +263,7 @@ perf-profile-creator-tests: create-performance-profile
 	hack/run-perf-profile-creator-functests.sh
 
 .PHONY: render-command-tests
-render-command-tests:
+render-command-tests: dist
```
Comment (fromanirh) [inline Makefile:266]: this should be a separate commit (a separate PR is not needed)

File: functests-render-command/1_render_command/render_test.go [hunk 2]
```diff
@@ -97,6 +97,11 @@ func runAndCompare(cmd *exec.Cmd) {
 		Expect(err).ToNot(HaveOccurred())
 
 		res := bytes.Compare(data, refData)
+		if res != 0 {
+			fmt.Fprintf(GinkgoWriter, "files: %q and %q are not identical\n",
+				filepath.Join(refPath, f.Name()),
+				filepath.Join(assetsOutDir, f.Name()))
```
Comment (fromanirh) [inline functests-render-command/1_render_command/render_test.go:103]: better, but I'd actually fix the expect right below.

File: testdata/render-expected-output/manual_tuned.yaml [hunk 3]
```diff
@@ -11,58 +11,7 @@ metadata:
...
...n# 100 tells the kernel to aggressively swap processes out of physical memory\n# and move them to swap cache\nvm.swappiness=10                              # latency-performance\n\n# The total time the scheduler will consider a migrated process\n# \"cache hot\" and thus less likely to be re-migrated\n# (system default is 500000, i.e. 0.5 ms)\nkernel.sched_migration_cost_ns=5000000        # latency-performance\n\n[selinux]\navc_cache_threshold=8192                      # Custom (atomic host)\n\n\n[net]\nnf_conntrack_hashsize=131072 \n\n\n[bootloader]\n# set empty values to disable RHEL initrd setting in cpu-partitioning \ninitrd_remove_dir=     \ninitrd_dst_img=\ninitrd_add_dir=\n# overrides cpu-partitioning cmdline\ncmdline_cpu_part=+nohz=on rcu_nocbs=${isolated_cores} tuned.non_isolcpus=${not_isolated_cpumask} intel_pstate=disable nosoftlockup\n\ncmdline_realtime=+tsc=nowatchdog intel_iommu=on iommu=pt isolcpus=managed_irq,${isolated_cores} systemd.cpu_affinity=${not_isolated_cores_expanded}\n\ncmdline_hugepages=+ default_hugepagesz=1G   hugepagesz=2M hugepages=128 \ncmdline_additionalArg=+ nmi_watchdog=0 audit=0 mce=off processor.max_cstate=1 idle=poll intel_idle.max_cstate=0 \n"
```
Comment (fromanirh) [inline testdata/render-expected-output/manual_tuned.yaml:14]: In the future we will likely need to update the code to emit manifests different from what we have today.
So the expected output will change, so the test will fail, what's the flow in these cases?

File: functests-render-command/1_render_command/render_test.go (hunk 2 above)
Comment (fromanirh) [reply functests-render-command/1_render_command/render_test.go:103]: let's keep it like this for now however, until we figure out a smarter diff.

Comment (fromanirh) [reply functests-render-command/1_render_command/render_test.go:103]: the idea in general is that the failing test output should tell us as precisely as possible what failed, instead of forcing us to to go review the expected output. Which is extra-hard because the YAMLs are (kinda-)minified.

File: testdata/render-expected-output/manual_tuned.yaml (hunk 3 above)
Comment (Tal-or) [reply testdata/render-expected-output/manual_tuned.yaml:14]: There are 2 options I see here:
1. the trivial way - fix the test
2. using regex to look for keywords that might indicate that something is wrong with the test output, so minor changes won't affect the test's outcome.
this approach is more flexible but yet less strict and might miss other things as well.

Comment (fromanirh) [reply testdata/render-expected-output/manual_tuned.yaml:14]: ok, let's keep it simple and let's fix the test. We will need to make the test  a bit easier to read on failures and easier to change (in a later PR).
